File: examples/ruby/stack_1.rb
class Stack
  def initialize
    @data = []
  end

  def push(x)
    @data << x
  end

  def pop
    @data.empty? ? nil : @data.pop
  end
end

s = Stack.new
s.push(1)
s.push(2)
s.push(3)
puts s.pop
puts s.pop

File: examples/python/stack_1.py
class Stack:
    def __init__(self):
        self._data = []

    def push(self, x):
        self._data.append(x)

    def pop(self):
        return self._data.pop() if self._data else None


if __name__ == "__main__":
    s = Stack()
    s.push(1)
    s.push(2)
    print(s.pop())
    print(s.pop())
